insta/src/macros.rs

```diff
@@ -19,12 +19,19 @@ macro_rules! _function_name {
 #[macro_export]
 macro_rules! _get_workspace_root {
     () => {{
-        use std::env;
+        use std::{env, option_env};
 
-        // Note the `env!("CARGO_MANIFEST_DIR")` needs to be in the macro (in
+        // Note the `option_env!("INSTA_WORKSPACE_ROOT")` and `env!("CARGO_MANIFEST_DIR")` needs to be in the macro (in
```

Comment:
But is this accurate? I don't think it needs to be in a macro?

And aren't we also referencing `INSTA_WORKSPACE_ROOT` in code above at complie time?

Reply:
One reason for doing this is to avoid embedding `CARGO_MANIFEST_DIR` in the produced binary, as this is build machine dependend and thus non repoducible.

If I don't do this, I would assume the string would still end up in the binary.

Maybe a different apporach would more suitable, maybe via conditional compile.

Reply:
Ah yes, I see.

But let's make the comment accurate (almost paste your reply into the code comment)

Reply:
OK, that makes sense; I suggested a change to the comments to ensure they remain accurate. Lmk if I'm mistaken!

Reply:
I was also thinking if I could add an actual test that checks if the binary contains a reference to `CARGO_MANIFEST_DIR`.

Reply:
I added a test that checks the binary for references to to the `CARGO_MANIFEST_DIR`